extensions/amp-app-banner/1.0/component/android.js

```diff
@@ -0,0 +1,104 @@
+import {docInfoService} from '#preact/services/document';
+import {platformService} from '#preact/services/platform';
+import {timerService} from '#preact/services/timer';
+import {urlService} from '#preact/services/url';
+import {viewerService} from '#preact/services/viewer';
+import {xhrService} from '#preact/services/xhr';
+
+import {user} from '#utils/log';
+
+import {openWindowDialog} from '../../../../src/open-window-dialog';
+
+const OPEN_LINK_TIMEOUT = 1500;
+
+/**
+ * @return {{openOrInstall: function(): void, promise: Promise<Response>}|null}
+ */
+export function getAndroidAppInfo() {
+  const win = self.window;
+  // We want to fallback to browser builtin mechanism when possible.
+  const isChromeAndroid =
+    platformService.isAndroid() && platformService.isChrome();
+  const isProxyOrigin = urlService.isProxyOrigin(win.location);
+  const canShowBuiltinBanner =
+    !isProxyOrigin && !viewerService.isEmbedded() && isChromeAndroid;
+
+  if (canShowBuiltinBanner) {
+    user().info(
```

Comment:
@rcebulko: Is user log available in Bento? I think this'll stop working as soon as we're out of the AMP mode.

Reply:
It shouldn't be, though last I checked other Bento code uses it. Personally I think we should get rid of almost all of the log system for Bento and either use `console` directly or a very thin wrapper around it.

Reply:
I'd be happy to drop this ... IMO a thin wrapper around `console` is a good choice, since direct `console` is often used for temporary/debugging code and should be eslint-prohibited.

Reply:
I'm seeing plenty of Bento code that uses `user()` and `dev()`, so I'm going to leave this as-is for this PR, and let this be addressed by a future discussion.